actions/get_hec_token.py

```diff
@@ -0,0 +1,83 @@
+import json
+import requests
+import urllib3
+urllib3.disable_warnings()
+import xml.dom.minidom
+
+from lib.base import SplunkBaseAction
+
+__all__ = [
+    'GetHecToken'
+]
+
+#
+class GetHecToken(SplunkBaseAction):
+    """Action to get HEC token"""
+    def __init__(self, config):
+        super(GetHecToken, self).__init__(config)
+        # Validate config is set
+        if config is None:
+            raise ValueError("No Splunk configuration details found")
+        if "splunk_instances" in config:
+            if config['splunk_instances'] is None:
+                raise ValueError("'splunk_instances' config empty.")
+            else:
+                pass
+        else:
+            raise ValueError("No Splunk configuration details found")
+
+    def run(self, instance):
+        """get hec token"""
+        """Collect instance details from config."""
+        # requests.packages.urllib3.disable_warnings()
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+        if not instance:
+            instance = "default"
+
+        # Build the base splunk url
+        splunk_config = self.config['splunk_instances'].get(instance)
+
+        host = splunk_config.get('host')
+        port = splunk_config.get('port')
+        scheme = splunk_config.get('scheme')
+        verify = splunk_config.get('verify')
+        username = splunk_config.get('username')
+        password = splunk_config.get('password')
+
+        base_url = "{}://{}:{}/".format(scheme, host, port)
+
+        if not instance:
+            instance = "default"
+
+        headers = {'Accept-Language': 'application/json'}
+
+        data = {
+          'name': 'stackstorm',
+          'index': 'main',
+          'indexes': 'main,summary'
+        }
+
+        response = requests.post(base_url + '/servicesNS/nobody/search/data/inputs/http',
```

Comment:
This does not appear to be able to work with the api token.  I'd recommend utilizing the `SplunkBaseAction` to perform all of the auth actions etc.  
The get_user.py is a good example of using the base class to do the hard for of auth, formatting, & etc.  https://github.com/StackStorm-Exchange/stackstorm-splunk/blob/master/actions/get_user.py.

Reply:
I'll go back and look at it. Using a base action is how I do all my other packs. When I first forked this, I was looking at send_log ation, which did not use a base action.